src/toolbar/toolbar-group.jsx

```diff
@@ -69,87 +130,20 @@ const ToolbarGroup = React.createClass({
     };
   },
 
-  //to update theme inside state whenever a new theme is passed down
-  //from the parent / owner using context
   componentWillReceiveProps(nextProps, nextContext) {
-    const newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
-    this.setState({muiTheme: newMuiTheme});
+    this.setState({
+      muiTheme: nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme,
+    });
   },
 
-  getTheme() {
-    return this.state.muiTheme.toolbar;
+  _handleMouseEnterFontIcon: (style) => (e) => {
```

Comment:
Yeah! That would be great. But it's going to need a lot of thought. I think an HOC that can handle these interactions would make this stateless (an a whole lot of other components) but I don't know how that is possible.